lib/redis/namespace_extensions.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class Redis
+  module NamespaceExtensions
+    def exists?(*args, &block)
+      call_with_namespace('exists?', *args, &block)
+    end
+  end
+end
+
+Redis::Namespace::COMMANDS['exists?'] = [:first]
+Redis::Namespace.prepend(Redis::NamespaceExtensions)
```

Comment:
Hmmm. I'm not sure I understand that. The fix in redis-namespace seems to be only adding `"exists?" => [:all]` to the namespaced commands? https://github.com/resque/redis-namespace/pull/171

Also, does using “exists?” change the Redis version requirements in any way?

Reply:
No, `exists?` isn't a Redis command, it's a Ruby command on the Redis library.

And no, the fix is not that, because at evaluation time the class iterates over that constant to `define_method` for each. If you just add another item to the constant, it won't define the method. But the addition to constant is also required because it controls the behaviour with the `:first` option.

Reply:
Alright, I can see the `define_method` thing. Shouldn't it be `= [:all]` though?

Reply:
I just used whatever the definition for `exists` was using.

Reply:
Alright, according to Redis' documentation, `EXISTS` can take multiple keys, so it should be `:all`, as the aforementioned PR does.

Reply:
Which PR was aforementioned?

Reply:
https://github.com/resque/redis-namespace/pull/171

Reply:
Ah, I see. Well, we never use that command with more than one argument.

Reply:
Does that mean we can never use it with more than one arguement?

Reply:
Yeah. Where were you going to use `Redis#exists?` with more than one argumeng?

Reply:
I don't know. Sounds like bad practice and stuff which could become confusing in the future.